sway-core/src/convert_parse_tree.rs

```diff
@@ -640,9 +684,13 @@ fn get_attributed_purity(
             purity = Purity::ReadsWrites;
         }
     };
-    match attributes.get(STORAGE_PURITY_ATTRIBUTE_NAME) {
-        Some(args) if !args.is_empty() => {
-            for arg in args {
+    match attributes.get(&AttributeKind::Storage) {
+        Some(attrs) if !attrs.is_empty() => {
+            for arg in attrs
+                .iter()
+                .filter(|attr| !attr.args.is_empty())
```

Comment:
This `filter` probably isn't necessary anyway, since `flat_map` would just skip it, I assume.